inference.py

```diff
@@ -0,0 +1,106 @@
+
+    # name
+    parser.add_argument('--checkname', type=str, default="_evaluation_crowdAI", help='set the checkpoint name')
+    parser.add_argument('--checkpoint-root', type=str, default='weights', help='the root directory of the checkpoints')
+
+    parser.add_argument('--resume', type=str, default="crowdAI", choices=["crowdAI", "spaceNet", "urban3d"], help='experiment to load')
+    parser.add_argument('--best-miou', action='store_true', default=False)
+
+    #boundaries
+    parser.add_argument('--incl-bounds', action='store_true', default=False, help='includes boundaries of masks in loss function')
+    parser.add_argument('--bounds-kernel-size', type=int, default=3, help='kernel size for calculating boundary')
+
+    parser.add_argument('--input-fn', type=str, required=True, help="path to an input GeoTIFF to run model on")
+    parser.add_argument('--output-fn', type=str, required=True, help="path to write predictions as a GeoTIFF")
+    parser.add_argument('--chip-size', type=int, default=1024, help="the size of input patches to sample from the input")
+    parser.add_argument('--padding', type=int, default=256, help="the number of pixels on each edge to downweight predictions to help prevent checkerboarding")
+    parser.add_argument('--switch-r-and-b', action='store_true', help='whether to switch the first and third bands of the input (for BGR formatted TIFFs)')
+
+
+    args = parser.parse_args()
+
+    args.dataset = "tile"
+    args.use_wandb = False
+
+    # Parse inputs
+    assert os.path.exists(args.input_fn)
+    assert not os.path.exists(args.output_fn)
+
+    args.cuda = not args.no_cuda and torch.cuda.is_available()
+    if args.cuda:
+        try:
+            args.gpu_ids = [int(s) for s in args.gpu_ids.split(',')]
+        except ValueError:
+            raise ValueError('Argument --gpu_ids must be a comma-separated list of integers only')
+
+    if args.sync_bn is None:
+        if args.cuda and len(args.gpu_ids) > 1:
+            args.sync_bn = True
+        else:
+            args.sync_bn = False
+
+    args.chip_stride = args.chip_size - 2 * args.padding
+    assert args.chip_size > 0
+
+    if args.test_batch_size is None:
+        args.test_batch_size = args.batch_size
+
+    if args.checkname is None:
+        args.checkname = 'deeplab-'+str(args.backbone)
+
+    # Run inference
+    torch.manual_seed(args.seed)
+    tester = Tester(args)
+
+    with rasterio.open(args.input_fn) as f:
+        input_width, input_height = f.width, f.height
+        input_profile = f.profile.copy()
+
+    output_hard = tester.inference(input_width, input_height)
+
+    input_profile["count"] = 1
+    input_profile["tiled"] = True
+    input_profile["compression"] = "lzw"
+    input_profile["predictor"] = 2
+
+    with rasterio.open(args.output_fn, "w", **input_profile) as f:
+        f.write(output_hard, 1)
+        f.write_colormap(1, {
+            0: (0, 0, 0, 0),
+            1: (255, 0, 0, 255)
+        })
```

Comment:
This is cool! Would the output TIFF file be viewable on any image viewing application? I've had troubles viewing TIFF files in plain image viewing apps and thus had to extract rasterizations to really see the image. It would be nice if I didn't have to use more code to view the output image.

Reply:
Yep this writes a georeferenced tiff (GeoTIFF). You can view this overlapped with the imagery in GIS applications like QGIS. E.g. this way you can make figures really easily:

![buildings_overlay](https://user-images.githubusercontent.com/1088078/140622598-a459db00-68eb-4d49-a2b0-f9df9327b0e9.jpg)